### PR TITLE
Confirm registration

### LINKED ISSUE #81: Initial registration
1. When I run `code-push register http://localhost:3000` my browser opens the login page, but I think it's supposed to open the register page.
2. Both auth/login and auth/register have no `action` defined on their forms, and thus the JavaScript code doesn't POST the data anywhere.
3. I added /users/registerCode as the action to the register form, but it does not return the token that is expected.

I am happy to help complete the implementation of this, but I don't want to duplicate efforts if you've already done it, and I want to make sure I'm doing the right thing. Thanks.

### PR DESCRIPTION
**Pull request by byronigoe**
Fixes #81

### CODE REVIEW

File: views/auth/confirm.pug [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            h4#mySmallModalLabel.modal-title Registration complete
+          .modal-body  Please proceed to login.
+          .modal-footer
+            button.btn.btn-default(type="button" data-dismiss="modal") Close
+            button#okBtn.btn.btn-primary(type="button" data-dismiss="modal") OK
+block js
+  script().
+    var submit = false;
+    $('#inputEmail').on('click', function () {
+      location.href = '/auth/register?email=' + $('#inputEmail').val();
```
Comment (rocwind) [inline views/auth/confirm.pug:35]: It feels surprise to see the page goes back when I clicked the disabled email input by accident...
Although it's a convenient way for user to change the registering email, I still tend to suggest we remove this function as user can always use the browser 'go back' for the purpose, what do you think?

**Comment by rocwind** [comment]
@byronigoe thanks for the contribution!
We didn't use the register function and didn't realize it's not implemented by the original author. The implementation looks great to me and I'd be happy to merge it in.
However, I have some security concerns about the public registertion (since it's not there before), would you mind to help adding a config item for enable/disable it? (e.g. `common.enableRegister`)

**Comment by byronigoe** [comment]
Okay. I've added a flag: `common.allowRegistration` which defaults to `false`.

**Comment by rocwind** [comment]
cool, thanks for the help!

**Comment by rocwind** [comment]
merge into master with commit messages rebased (coventional commit style for auto-generate changelog)

https://github.com/shm-open/code-push-server/commit/b8b0276f4a54199ce66fb646b327d323a434c7cc
https://github.com/shm-open/code-push-server/commit/53ffab6a2b18abd9468a87370c2300bf27fc27ec

**Pull request closed without merging**